subsys/bootloader/c_runtime_setup/CMakeLists.txt

```diff
@@ -47,6 +46,27 @@ target_include_directories_ifdef(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_ZEPHYR
   ${ARM_INC}/include/cortex_m
   )
 
+if(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_ZEPHYR)
+set(INIT_O_PATH ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/kernel_init.dir/${ZEPHYR_BASE}/kernel/init.c.obj)
+target_link_libraries_ifdef(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_ZEPHYR
+  c_runtime_setup
+  PRIVATE
+  ${INIT_O_PATH}.mod
+)
+
+add_library(kernel_init STATIC ${ZEPHYR_BASE}/kernel/init.c)
+target_link_libraries(kernel_init zephyr_interface)
+
+add_custom_command(
+  OUTPUT ${INIT_O_PATH}.mod
+  COMMAND ${CMAKE_OBJCOPY} --redefine-sym _Cstart=UNUSED_Cstart ${INIT_O_PATH} ${INIT_O_PATH}.mod
```

Comment:
I googled around and found the `-u` option for `ld`.

https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html

Do you think it could be used in this case?

Reply:
Looks promising, can't say for sure without testing it but I think we only run the linker once for the executable and not for the libraries(correct me if I'm wrong) so it could be hard to force the specific library to have the symbol undefined.

Reply:
Tried to add `-Wl,-u,_Cstart` to the compilation of the library didn't work. since it  only "Pretend the symbol symbol is undefined, to force linking of library modules to define it. You can use -u multiple times with different symbols to force loading of additional library modules." when the library is built it is still defined. So the way the build-system is setup we link `libc_runtime_setup` into the target `libbootloader` so if we do this at that step we will undefine the `_Cstart` symbol in both or only in `libbootloader` this will then force linking against the `_Cstart` function which we did not want. So I couldn't find a solution to this.

Reply:
Can it work if you separate files into more libs?

Reply:
Won't solve the problem as `bootloader.c` is the top level we are linking against.

Reply:
AFAICT this works:

https://github.com/SebastianBoe/fw-nrfconnect-nrf/commit/29df246824fece9e4c8155124fa828e4cbe9c756

Essentially, instead of changing the symbol after it has been compiled we have the preprocessor replace the _Cstart token with _Cstart_UNUSED before it is compiled.

Reply:
Can confirm this works for replacing the `_Cstart` token and removes it from the symbol table. Since we don't really need it I'll keep this approach thanks @SebastianBoe